Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIPermissionsCocoa.mm

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if !__has_feature(objc_arc)
+#error This file requires ARC. Add the "-fobjc-arc" compiler flag for this file.
+#endif
+
+#import "config.h"
+#import "WebExtensionAPIPermissions.h"
+
+#import "WebExtensionContextMessages.h"
+#import "WebProcess.h"
+#import <wtf/cocoa/VectorCocoa.h>
+
+#if ENABLE(WK_WEB_EXTENSIONS)
+
+namespace WebKit {
+
+static NSString *permissionsKey = @"permissions";
+static NSString *originsKey = @"origins";
+
+void WebExtensionAPIPermissions::getAll(WebPage* page, Ref<WebExtensionCallbackHandler>&& callback)
+{
+    RELEASE_LOG(Extensions, "permissions.getAll()");
+
+    WebProcess::singleton().sendWithAsyncReply(Messages::WebExtensionContext::GetAll(page->webPageProxyIdentifier()), [ protectedThis = Ref { *this }, callback = WTFMove(callback) ](Vector<String> permissions, Vector<String> origins) {
+        callback->call(@{
+            permissionsKey: createNSArray(permissions).get(),
+            originsKey: createNSArray(origins).get()
+        });
+    }, extensionContext().identifier().toUInt64());
+}
+
+void WebExtensionAPIPermissions::contains(WebPage* page, NSDictionary *details, Ref<WebExtensionCallbackHandler>&& callback, NSString **)
+{
+    RELEASE_LOG(Extensions, "permissions.contains()");
+
+    HashSet<String> permissions, origins;
+    parseDetailsDictionary(details, permissions, origins);
+
+    WebProcess::singleton().sendWithAsyncReply(Messages::WebExtensionContext::Contains(page->webPageProxyIdentifier(), permissions, origins), [ protectedThis = Ref { *this }, callback = WTFMove(callback) ](bool containsPermissions) {
```

Comment:
Drop the spaces inside the brackets: `[protectedThis = Ref { *this }, callback = WTFMove(callback)]`